packages/calcite-components/src/components/dialog/dialog.tsx

```diff
@@ -0,0 +1,509 @@
+                  <slot name={SLOTS.footerStart} slot={PANEL_SLOTS.footerStart} />
+                  <slot name={SLOTS.footerEnd} slot={PANEL_SLOTS.footerEnd} />
+                </slot>
+                <slot />
+              </calcite-panel>
+            </slot>
+          </div>
+        </div>
+      </Host>
+    );
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private Properties/ State
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el: HTMLCalciteDialogElement;
+
+  @State() opened = false;
+
+  @State() hasFooter = true;
+
+  @State() hasContentTop = false;
+
+  @State() hasContentBottom = false;
+
+  @State() effectiveLocale: string;
+
+  @Watch("effectiveLocale")
+  effectiveLocaleChange(): void {
+    updateMessages(this, this.effectiveLocale);
+  }
+
+  @State() defaultMessages: DialogMessages;
+
+  openTransitionProp = "opacity";
+
+  transitionEl: HTMLDivElement;
+
+  focusTrap: FocusTrap;
+
+  private panelEl: HTMLCalcitePanelElement;
+
+  private ignoreOpenChange = false;
+
+  private mutationObserver: MutationObserver = createObserver("mutation", () =>
+    this.handleMutationObserver(),
+  );
+
+  //--------------------------------------------------------------------------
+  //
+  //  Event Listeners
+  //
+  //--------------------------------------------------------------------------
+
+  @Listen("keydown", { target: "window" })
+  handleEscape(event: KeyboardEvent): void {
+    if (this.open && event.key === "Escape" && !event.defaultPrevented) {
+      this.open = false;
+      event.preventDefault();
+    }
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  //--------------------------------------------------------------------------
+  /** Fires when the component is requested to be closed and before the closing transition begins. */
+  @Event({ cancelable: false }) calciteDialogBeforeClose: EventEmitter<void>;
+
+  /** Fires when the component is closed and animation is complete. */
+  @Event({ cancelable: false }) calciteDialogClose: EventEmitter<void>;
+
+  /** Fires when the component is added to the DOM but not rendered, and before the opening transition begins. */
+  @Event({ cancelable: false }) calciteDialogBeforeOpen: EventEmitter<void>;
+
+  /** Fires when the component is open and animation is complete. */
+  @Event({ cancelable: false }) calciteDialogOpen: EventEmitter<void>;
+
+  /** Fires when the content is scrolled. */
+  @Event({ cancelable: false }) calciteDialogScroll: EventEmitter<void>;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  //--------------------------------------------------------------------------
+
+  /**
+   * Sets focus on the component's "close" button (the first focusable item).
```

Comment:
I didn't try. Do we know why modal wasn't using delegatesFocus?

Reply:
@benelan do you remember the cons of using delegatesFocus and why we weren't using it in all components? We should probably document this somewhere for future reference. cc @jcfranco incase he knows.

Reply:
I believe Modal was skipped because we needed to use custom logic for the focus trap. I'll try to refresh my memory and add some documentation to the wiki after the release. I'd say let's move forward without delegatesFocus for now, and I can re-evaluate after I'm refreshed.

Reply:
I found the issue: #514 

I was putting research notes in the body, including a list of components and why they were skipped. Modal (and most of the skipped components) were related to the `disabled` property:

> If a "mock disabled" element is first in the flat tree, it will be delegated focus. A lot of our components can have slotted content as the first focusable element.

Reply:
Correction, the important info is in https://github.com/Esri/calcite-design-system/issues/5147#issuecomment-1349616672

I only thing I added to the body was **bolded** responses to Franco's task list.

Reply:
nice find! we should save this somewhere in a markdown or wiki page.

Reply:
One more downside to using `delegatesFocus` is that it prevents blurring any focusable element that has focus by clicking on a non-focusable area within the `delegatesFocus` host (see [example](https://codepen.io/jcfranco/pen/PorGwMj)). This might not be desired in [some scenarios](https://github.com/Esri/calcite-design-system/pull/9701#discussion_r1664651315).